galera/src/replicator_str.cpp

```diff
@@ -829,13 +829,27 @@ ReplicatorSMM::request_state_transfer (void* recv_ctx,
             (void)ist_receiver_.finished();
         }
     }
+    else if (sst_req_len == 0)
```

Comment:
Let me see if I understand this case correctly.
sst_req_len = 0 means there is no need of SST but IST would suffice.
Now if server crashes in middle of IST then we need to ensure that state is set to UNDEFINED for full SST on restart.
As per latest change in recv_IST function if there is SST before IST then SST routine takes care of setting state = UNDEFINED but if there is no SST before IST then existing recv_IST newly added loops will take care of the same.

Also, may be it is worth documenting this scenario in form of comment.